include/ofi_hmem.h

```diff
@@ -121,8 +121,8 @@ struct ofi_hmem_ops {
 	int (*copy_from_hmem)(uint64_t device, void *dest, const void *src,
 			      size_t size);
 	bool (*is_addr_valid)(const void *addr, uint64_t *device, uint64_t *flags);
-	int (*get_handle)(void *dev_buf, void **handle);
-	int (*open_handle)(void **handle, uint64_t device, void **ipc_ptr);
+	int (*get_handle)(void *dev_buf, size_t size, void **handle);
+	int (*open_handle)(void **handle, size_t size, uint64_t device, void **ipc_ptr);
```

Comment:
Separate these changes into its own patch.

@aingerson @j-xiong @wzamazon  - According to the ROCR documentation, hsa_amd_ipc_memory_create() (i.e. get_handle() only succeeds of the buffer is the fully allocated buffer.  dev_buf must point to the start of the buffer and len must be the allocation.  Does this same restriction hold for the other hmem interfaces?

Maybe more importantly, do we expect applications will adhere to this restriction when performing transfers?  I'm skeptical that this is the case.  And if it isn't, does the ROCR implementation need to adjust for the buffers accessed by the application, versus those which must be passed into the hsa_ calls?

Reply:
ofi_hmem_get_base_addr() returns the length of the buffer. This is then passed to the get_handle() call. Then passed on to the other process, which uses it in the open_handle()

The ROCR version of get_base_addr() will return the entire buffer size allocated by the user. For example if you allocate a  1M buffer but the application passes a pointer within that buffer, then base_addr will return the base_addr of the original buffer + the entire length of the buffer.

this information + offsets can be used in the caching.

I'm not sure what you mean when you refer to the restrictions.

Reply:
From onaAPI level-zero document:  `The pointer must be base pointer of the device memory allocation; i.e. the value returned from [zeMemAllocDevice]`.  CUDA is silimar `Takes a pointer to the base of an existing device memory allocation created with cudaMalloc and exports it for use in another process.`

The handle always refers to the entire allocation, so I don't think the `size` parameter is needed. If ROCR requires one, it can always be obtained via the `get_base_addr` method. There is no need to change the API here.

Reply:
I did get it via get_base_addr(), that's already in format_ipc(). However, if I don't pass it then I'll need to call it twice. Not sure if this is necessary.

Reply:
I don't call it. It was added as part of the caching patches. It's being called from smr_format_ipc(). It's called to get the base address and the base length, then the base address is being passed to ofi_hmem_get_handle() to get the ipc_handle